cli/popper/commands/cmd_run.py

```diff
@@ -91,10 +91,8 @@ def run_pipeline(project_root, pipeline, timeout, skip):
 
     for stage in pipeline['stages']:
 
-        if os.path.isfile(stage):
-            stage_file = stage
-        elif os.path.isfile(stage + '.sh'):
-            stage_file = stage + '.sh'
+        if pu.get_filename(abs_path, stage):
+            stage_file = pu.get_filename(abs_path, stage)
```

Comment:
this can avoid invoking `get_filename` twice by doing:

```python
stage_file = pu.get_filename(abs_path, stage)
skipped = skip.split(',')

if not stage_file or stage in skipped:
    continue
```